src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+    }
+
+    @Override
+    public BreakpointEvidence next() {
+        if ( !hasNext() ) {
+            throw new NoSuchElementException("No next element.");
+        }
+        return listItr.next();
+    }
+
+    private boolean isValidated( final List<BreakpointEvidence> evList ) {
+        for ( final BreakpointEvidence ev : evList ) {
+            if ( ev.isValidated() ) return true;
+        }
+        return false;
+    }
+
+    boolean anyPassesFilter(final List<BreakpointEvidence> evidenceList) {
+        for(final BreakpointEvidence evidence : evidenceList) {
+            final double evidenceGoodProbability = predictor.predictSingle(getFeatures(evidence));
+            if(evidenceGoodProbability > thresholdProbability) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Compute features vector for a piece of BreakpointEvidence
+     */
+    @VisibleForTesting
+    EvidenceFeatures getFeatures(final BreakpointEvidence evidence) {
+        // create new struct for these two, use CigarOperator to update if instanceof ReadEvidence
+        final CigarQualityInfo cigarQualityInfo = new CigarQualityInfo(evidence);
+        final double evidenceType = evidenceTypeMap.get(evidence.getClass().getSimpleName());
+        final double mappingQuality = (double)getMappingQuality(evidence);
+        final double templateSize = getTemplateSize(evidence);
+
+        // calculate these similar to BreakpointDensityFilter, but always calculate full totals, never end early.
+        final CoverageScaledOverlapInfo individualOverlapInfo = getIndividualOverlapInfo(evidence);
+        final CoverageScaledOverlapInfo clusterOverlapInfo = getClusterOverlapInfo(evidence);
+
+        // calculate properties related to overlap of intervals on the reference genome
+        final double referenceGapOverlap = getGenomeIntervalsOverlap(evidence, genomeGaps);
+        final double umapS100 = getGenomeIntervalsOverlap(evidence, umapS100Mappability);
+
+        return new EvidenceFeatures(
+            new double[]{
+                cigarQualityInfo.basesMatched, cigarQualityInfo.referenceLength, evidenceType, mappingQuality, templateSize,
+                individualOverlapInfo.numOverlap, individualOverlapInfo.overlapMappingQuality,
+                individualOverlapInfo.meanOverlapMappingQuality, individualOverlapInfo.numCoherent,
+                individualOverlapInfo.coherentMappingQuality,
+                clusterOverlapInfo.numOverlap, clusterOverlapInfo.overlapMappingQuality,
+                clusterOverlapInfo.meanOverlapMappingQuality, clusterOverlapInfo.numCoherent,
+                clusterOverlapInfo.coherentMappingQuality,
+                referenceGapOverlap, umapS100
+            }
+        );
+    }
+
+    private int getMappingQuality(final BreakpointEvidence evidence) {
+        // Note: return "max" mapping quality for non-ReadEvidence. Reasoning: some features depend on sum or average of
```

Comment:
The classifier can handle missing values. Reasons I didn't set this to NaN:
  1. The features involving sum or mean of mapping quality for overlapping evidence have fairly high importance (overlap_mapping quality and mean_overlap_mapping_quality are more important than num_coherent), and would become problematic when allowing mapping quality to be NaN.
  2. Generally the non-read evidence did seem to have high weight in the Bayesian approaches
  3. The non-read evidence at this point seems like it's inherently well localized (to my understanding it's a change in distribution of read pile-up at a certain point in the reference, so I thought the likelihood of the non-read evidence being in the right *location* was high).

I'll think hard about whether it's possible to replace the "overlap quality" type features with something similar that allows for non-read evidence to have NaN mapping qualities, which I agree is logically more defensible. Right now it would be difficult to switch without substantial redesign of the features.